pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/oauth2/AuthenticationOAuth2Builder.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.client.impl.auth.oauth2;
+
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import org.apache.pulsar.client.api.Authentication;
+
+/**
+ * Builder for {@link AuthenticationOAuth2} class.
+ */
+public final class AuthenticationOAuth2Builder {
+
+    /**
+     * Create a builder instance.
+     *
+     * @return builder
+     */
+    public static AuthenticationOAuth2Builder builder() {
+        return new AuthenticationOAuth2Builder();
+    }
+
+    private double earlyTokenRefreshPercent = AuthenticationOAuth2.EARLY_TOKEN_REFRESH_PERCENT_DEFAULT;
+    private ScheduledThreadPoolExecutor scheduler;
+    private ClientCredentialsConfiguration clientCredentialsConfiguration;
+
+    /**
+     * Set the {@link ClientCredentialsConfiguration} when using the OAuth2 client credentials flow.
+     * @return builder
+     */
+    public AuthenticationOAuth2Builder setClientCredentialsConfiguration(
```

Comment:
In Pulsar, builders usually don't use `set*` naming convention. it would be good to follow a similar naming convention that has been used in other parts of Pulsar. for example, rename `setClientCredentialsConfiguration` -> `clientCredentialsConfiguration`.